flink-runtime/src/test/java/org/apache/flink/runtime/taskexecutor/TaskExecutorPartitionLifecycleTest.java

```diff
@@ -0,0 +1,279 @@
+	}
+
+	private void testPartitionRelease(
+		TriConsumer<JobID, ResultPartitionID, TaskExecutorGateway> releaseAction,
+		BiConsumer<JobID, JobAwareShuffleEnvironmentImpl> postReleaseActionAssertion) throws Exception {
+
+		final TaskManagerLocation taskManagerLocation = new LocalTaskManagerLocation();
+
+		final TaskSlotTable taskSlotTable = new TaskSlotTable(
+			Collections.singletonList(ResourceProfile.UNKNOWN),
+			new TimerService<>(TestingUtils.defaultExecutor(), timeout.toMilliseconds()));
+
+		final JobLeaderService jobLeaderService = new JobLeaderService(taskManagerLocation, RetryingRegistrationConfiguration.defaultConfiguration());
+
+		final TaskExecutorLocalStateStoresManager localStateStoresManager = new TaskExecutorLocalStateStoresManager(
+			false,
+			new File[]{tmp.newFolder()},
+			Executors.directExecutor());
+
+		final JobAwareShuffleEnvironmentImpl<?> jobAwareShuffleEnvironment = new JobAwareShuffleEnvironmentImpl<>(new NettyShuffleEnvironmentBuilder().build());
+
+		final TaskManagerServices taskManagerServices = new TaskManagerServicesBuilder()
+			.setTaskManagerLocation(taskManagerLocation)
+			.setTaskSlotTable(taskSlotTable)
+			.setJobLeaderService(jobLeaderService)
+			.setTaskStateManager(localStateStoresManager)
+			.setShuffleEnvironment(jobAwareShuffleEnvironment)
+			.build();
+
+		final CompletableFuture<Void> taskFinishedFuture = new CompletableFuture<>();
+
+		final TestingJobMasterGateway jobMasterGateway = new TestingJobMasterGatewayBuilder()
+			.setRegisterTaskManagerFunction((s, location) -> CompletableFuture.completedFuture(new JMTMRegistrationSuccess(ResourceID.generate())))
+			.setOfferSlotsFunction((resourceID, slotOffers) -> CompletableFuture.completedFuture(slotOffers))
+			.setUpdateTaskExecutionStateFunction(taskExecutionState -> {
+				if (taskExecutionState.getExecutionState() == ExecutionState.FINISHED) {
+					taskFinishedFuture.complete(null);
+				}
+				return CompletableFuture.completedFuture(Acknowledge.get());
+			})
+			.build();
+
+		final TestingTaskExecutor taskManager = createTestingTaskExecutor(taskManagerServices, new HeartbeatServices(Long.MAX_VALUE, Long.MAX_VALUE));
+
+		try {
+			taskManager.start();
+			taskManager.waitUntilStarted();
+
+			final String jobMasterAddress = "jm";
+			rpc.registerGateway(jobMasterAddress, jobMasterGateway);
+
+			// inform the task manager about the job leader
+			jobLeaderService.addJob(jobId, jobMasterAddress);
+			jobManagerLeaderRetriever.notifyListener(jobMasterAddress, UUID.randomUUID());
+
+			while (!taskManagerServices.getJobManagerTable().contains(jobId)) {
+				Thread.sleep(50);
+			}
+
+			final ExecutionAttemptID eid1 = new ExecutionAttemptID();
+			final IntermediateResultPartitionID partitionId = new IntermediateResultPartitionID();
+			final ResultPartitionID resultPartitionId = new ResultPartitionID(partitionId, eid1);
+			final ResultPartitionDeploymentDescriptor task1ResultPartitionDescriptor =
```

Comment:
`ResultPartitionDeploymentDescriptor` generation might reuse `ResultPartitionTest.createPartitionDeploymentDescriptor(ResultPartitionType.BLOCKING)`, then `partitionId` and `eid1` could also be got from created `RPDD`.